core/src/main/java/io/micronaut/core/annotation/AnnotationMetadata.java

```diff
@@ -611,6 +616,97 @@ default boolean isDeclaredAnnotationPresent(@Nonnull Class<? extends Annotation>
         return result.map(OptionalInt::of).orElseGet(OptionalInt::empty);
     }
 
+    /**
+     * The value as an {@link OptionalInt} for the given annotation and member.
+     *
+     * @param annotation The annotation
+     * @param member     The member
+     * @return THe {@link OptionalInt} value
+     */
+    default @Nonnull OptionalInt intValue(@Nonnull Class<? extends Annotation> annotation, @Nonnull String member) {
+        ArgumentUtils.requireNonNull("annotation", annotation);
+        return intValue(annotation.getName(), member);
+    }
+
+    /**
+     * The value as an {@link OptionalInt} for the given annotation and member.
+     *
+     * @param annotation The annotation
+     * @return THe {@link OptionalInt} value
+     */
+    default @Nonnull OptionalInt intValue(@Nonnull Class<? extends Annotation> annotation) {
+        ArgumentUtils.requireNonNull("annotation", annotation);
+        return intValue(annotation, VALUE_MEMBER);
+    }
+
+    /**
+     * The value as an optional string for the given annotation and member.
+     *
+     * @param annotation The annotation
+     * @param member     The member
+     * @return The string value if it is present
+     */
+    default @Nonnull Optional<String> stringValue(@Nonnull String annotation, @Nonnull String member) {
+        ArgumentUtils.requireNonNull("annotation", annotation);
+        ArgumentUtils.requireNonNull("member", member);
+
+        return getValue(annotation, member, String.class);
+    }
+
+    /**
+     * The value as an optional string for the given annotation and member.
+     *
+     * @param annotation The annotation
+     * @param member     The member
+     * @return The string value if it is present
+     */
+    default @Nonnull Optional<String> stringValue(@Nonnull Class<? extends Annotation> annotation, @Nonnull String member) {
+        ArgumentUtils.requireNonNull("annotation", annotation);
+        return stringValue(annotation.getName(), member);
+    }
+
+    /**
+     * The values as string array for the given annotation and member.
+     *
+     * @param annotation The annotation
+     * @param member     The member
+     * @return The string values if it is present
+     */
+    default @Nonnull String[] stringValues(@Nonnull Class<? extends Annotation> annotation, @Nonnull String member) {
+        return StringUtils.EMPTY_STRING_ARRAY;
```

Comment:
Why doesn't this implementation default to getting the values the old way?

Reply:
Could do, but the old ways performance is not good. This method is overridden anyway be the implementation. I put this in there for backwards compatibility so if someone had implemented this interface (unlikely) it would still compile.